### PR TITLE
reading stats - adjusted progress bars, added spaced practice

### PR DESCRIPTION
**Pull request by darrenmason**
image

This PR renders progress bars for course overall, current_pages and spaced_pages.  It is currently not rendering periods.  Percent change is calculated on spaced_pages based on previous_attempt.

current issues:

1) the way either bootstrap progressbar or react-bootstrap calculates percentage width is causing smaller segments to overflow:

image

2) the overall course stats needs stat: 'not started' and/or an accurate total_count to render the progress bar properly.

### CODE REVIEW

File: src/components/task-plan/reading-stats.cjsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+Router = require 'react-router'
+
+{TaskPlanStore, TaskPlanActions} = require '../../flux/task-plan'
+
+Stats = React.createClass
+  mixins: [Router.State, Router.Navigation]
+
+  getInitialState: ->
+    id = @getParams().id
+    if (id)
```
Comment (philschatz) [inline src/components/task-plan/reading-stats.cjsx]: I think the reading stats only apply to an existing reading so the following code is never executed

File: src/components/task-plan/reading-stats.cjsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+          <BS.ProgressBar className="reading-progress-bar" bsStyle="success" label="%(now)s%" now={@percent(data.correct_count,data.correct_count+data.incorrect_count)} key={1} />
+        </BS.ProgressBar>
+        <div className="reading-progress-delta">{@percentDelta(data.correct_count,data.previous_attempt.correct_count)}% change</div>
+      </div>
+    </div>
+
+  render: ->
+    id = @getPlanId()
+
+    if TaskPlanStore.isLoaded(id)
```
Comment (philschatz) [inline src/components/task-plan/reading-stats.cjsx]: Did you try using the Loadable mixin to handle this case (and the `componentWill(Un)Mount`)?

Comment (darrenmason) [inline src/components/task-plan/reading-stats.cjsx]: yes, changed to loadable-mixin

File: src/components/task-plan/reading-stats.cjsx (hunk 1 above)
Comment (darrenmason) [inline src/components/task-plan/reading-stats.cjsx]: removed

Comment (philschatz) [inline src/components/task-plan/reading-stats.cjsx]: Is it removed? This comment should disappear if the text was removed.

File: style/reading-stats.less [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+
+// debug:
+
+.progress{
+  overflow:visible;
+}
+
+.progress-bar {
+    // overflow-x:visible;
```
Comment (philschatz) [inline style/reading-stats.less]: Are these for documentation? If not, they should probably be removed.

**Comment by Fredasaurus** [comment]
Cool, didn't know we were working on this yet, thought it was a sprint 9 task.

Have a design for this here. http://fredasaurus.github.io/tutor-designs/calendar/calendar.html

screen shot 2015-03-19 at 1 56 09 pm

**Comment by philschatz** [comment]
Thanks! @darrenmason : I copied your comment into the Pull Request description field.

**Comment by Fredasaurus** [comment]
A couple comments:
1. We're not sure about how the first section will be visualized, working on that still. May be a pie chart.
2. Probably not going to use SPARFA with %delta for spaced practice, not accurate enough. MOst likely will be right / wrong.

**Comment by darrenmason** [comment]
Eventually, will stats view need to be rendered both in its own page and in a modal?  If so will those two layouts be different, or could they pull from the same view?

**Comment by Fredasaurus** [comment]
1. Yes Here, click on "Read Chapter 4" 
http://mockups.openstax.org/Tutor_MVP_v2/start.html#p=course_calendar_-_teacher
and here
http://mockups.openstax.org/Tutor_MVP_v2/start.html#p=reading_review_metrics_-_class
2. The view should be the same.

**Comment by philschatz** [comment]
currently there is no modal (since there is no calendar) but for the sprint the BE was tasked with making the stats and we needed a place to show them so I opted for `/readings/1/stats`

**Comment by Fredasaurus** [comment]
Also, I just used a bs modal for demo purposes.

**Pull request merged**